src/utils/error.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * This file contains extended errors for the application.
+ */
+
+export class ApplicationError extends Error {
+    public code: number;
+
+    constructor(message?: string, code?: number) {
+        super();
+
+        Error.captureStackTrace(this, this.constructor);
+        this.name = this.constructor.name;
+        this.message = message || "unknown application error";
+        this.code = code || 500;
+    }
+}
+
+export class ServerError extends ApplicationError {
+    constructor(message?: string, code?: number) {
+        super(message || "server side error", code || 500);
+    }
+}
+
+export class UserNotFoundError extends ServerError {
```

Comment:
```suggestion
export class UserNotFoundError extends ClientError {
```

Reply:
The user may exist but we our service maybe down so it could be a ServerError as much as a ClientError. Therefor, I suggest: `export class UserNotFoundError extends ApplicationError {`

Reply:
Ok, do that